android/src/main/kotlin/co/doneservices/callkeep/CallKeepNotificationManager.kt

```diff
@@ -0,0 +1,439 @@
+        }
+
+        override fun onPrepareLoad(placeHolderDrawable: Drawable?) {
+        }
+    }
+
+
+    fun showIncomingNotification(data: Bundle) {
+        data.putLong(EXTRA_TIME_START_CALL, System.currentTimeMillis())
+
+        notificationId = data.getString(EXTRA_CALLKEEP_ID, "callkeep").hashCode()
+        createNotificationChanel(
+            data.getString(EXTRA_CALLKEEP_INCOMING_CALL_NOTIFICATION_CHANNEL_NAME, "Incoming Calls"),
+            data.getString(EXTRA_CALLKEEP_MISSED_CALL_NOTIFICATION_CHANNEL_NAME, "Missed Calls"),
+        )
+
+        notificationBuilder = NotificationCompat.Builder(context, NOTIFICATION_CHANNEL_ID_INCOMING)
+        notificationBuilder.setAutoCancel(false)
+        notificationBuilder.setChannelId(NOTIFICATION_CHANNEL_ID_INCOMING)
+        notificationBuilder.setDefaults(NotificationCompat.DEFAULT_VIBRATE)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            notificationBuilder.setCategory(NotificationCompat.CATEGORY_CALL)
+            notificationBuilder.priority = NotificationCompat.PRIORITY_MAX
+        }
+        notificationBuilder.setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+        notificationBuilder.setOngoing(true)
+        notificationBuilder.setWhen(0)
+        notificationBuilder.setTimeoutAfter(data.getLong(EXTRA_CALLKEEP_DURATION, 0L))
+        notificationBuilder.setOnlyAlertOnce(true)
+        notificationBuilder.setSound(null)
+        notificationBuilder.setFullScreenIntent(
+                getActivityPendingIntent(notificationId, data), true
+        )
+        notificationBuilder.setContentIntent(getActivityPendingIntent(notificationId, data))
+        notificationBuilder.setDeleteIntent(getTimeOutPendingIntent(notificationId, data))
+        val hasVideo = data.getBoolean(EXTRA_CALLKEEP_HAS_VIDEO, false)
+        val notificationIcon = data.getString(EXTRA_CALLKEEP_NOTIFICATION_ICON, "")   
+        if(notificationIcon.isEmpty()){
+        var smallIcon = context.applicationInfo.icon
+            if (hasVideo) {
+                smallIcon = R.drawable.ic_video
+            } else {
+                if (smallIcon >= 0) {
+                    smallIcon = R.drawable.ic_accept
+                }
+            }
+            notificationBuilder.setSmallIcon(smallIcon)
+        } else {
+            val identifier = context.resources.getIdentifier(notificationIcon, "drawable", context.packageName)
+            val icon = IconCompat.createWithResource(context, identifier)
+            notificationBuilder.setSmallIcon(icon)
+        }
+
+        val accentColor = data.getString(EXTRA_CALLKEEP_ACCENT_COLOR, "#4CAF50")
+        try {
+            notificationBuilder.color = Color.parseColor(accentColor)
+        } catch (error: Exception) {
+        }
+        notificationBuilder.setChannelId(NOTIFICATION_CHANNEL_ID_INCOMING)
+        notificationBuilder.priority = NotificationCompat.PRIORITY_MAX
```

Comment:
This is already done further up behind a `Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP` check